toggleToolbox.m

```diff
@@ -179,20 +179,18 @@
 
     % Initialize tb states structure
     if ~restoremode
-        if exist(storefile,'file')==2
-            % Load previous paths and toggle states, if any
-            S = load(storefile);
-            toolbox_states = S.toolbox_states;
-            clear S;
-        else
-            % Otherwise, just initialize it by storing the previous
-            % path strings...
+        % Load previous paths and toggle states, if any
+        toolbox_states=getpref('RodyOldenhuis',...%author name as group ID
+            ['toggleToolbox___',...%function group
+            'toolbox_states'],...%preference name
+            []);

```

Comment:
Hmmm...though I'm not against name branding, I _do_ think this is too much...How about: 

```suggestion
        toolbox_states = getpref('toggleToolbox___',...            
                                 'toolbox_states',...
                                 []);
```

Reply:
I got the idea from Yair Altman, who uses his initials. Using your name has multiple advantages.

1. Name recognition. It is a clear mark of ownership over whose data should be stored here.
1. Decreasing the chance of name collisions. If someone else decides to make a toggleToolbox function they will not be using this pref group.
1. Allow you flexibility. If you ever decide to change the function name or (more likely) write another function that accesses this same struct you will not have an old function name as the pref group. You will still have that name in your pref name, but that is why the comment is talking about function group.

These don't require the group name to be your full name, but are equally valid with any other string that uniquely identifies you.

The three underscores serve as a separator, so if you want to use the function as the field name, I would suggest removing those underscores.

Reply:
...I guess this is a bit of a pet peeve of mine; reading back I realized I rambled on for a bit too long...Sorry 'bout that 😊 

Some counterarguments: 

1. My problem is that this is (in principle) visible from outside of the toolbox. Imagine that all The MathWorks employees used their own initials for this sort of thing, and then getting an error due to a collision with "John Smith" from TMW/UI design. Imagine the Linux syslog having entries like `LTORVALDS: running modprobe` or so; imagine the whole syslog littered with his name and many thousands of other names...It just does not add anything useful, rather, it _detracts_ quite a bit from the quality and usability. Moreover, if _you_ merge into this repository, I don't regard it as "my" toolbox anymore; it's become "ours". I would _at most_ expect the company/team name here, but this sort of info is really better off in documentation/license/project management/anything else _separate_ from the code. 

2. Like above; plenty of people are called "John Smith" or "Jane Johnson" 😊 There's _ample_ chance to collide when using initials/personal names; ask any sysadmin who started out giving email addresses in the `lastname@company` format. Also, with the underscores, we're pretty safe; leading/trailing underscores signal to me "internal use for tool \<insert name here\>"; a name doesn't effectively convey that.  

3. That's a trivial IDE task ("replace all occurrences"). The same logic holds for error/warning IDs (which, admittedly, I always write like `error([mfilename() ':error_tag'], 'Error message');`, which is a snippet that was born out of the severely lacking MATLAB IDE...but that's another story). If you'd use my name, you can't `grep` for it anymore...which brings me to:

4. Forking. Whenever someone forks our repository and does their own thing with it, the name tag has now become **false** branding; even if they'd change 99% of the code, my name would still be there, for better or for worse. A fork is not _mine_, it's _theirs_. Using names for this just makes it a huge pain for anyone else to take over ownership, which in my view kind of rubs against open-source philosophies. 

In general, I'd really like to keep the _who-dunnit_ separate from the code. Names/authors/etc. are already protected by the BSD license; don't annoy everyone with it, all the time, all over the place.

Reply:
Maybe the reason is that I don't see this function as a toolbox. Then it would make sense to me to use the toolbox name as the group name and as the first element in your error tag (as Mathworks is doing as well). Next time I think will change the comment to "author name or toolbox name as group ID".

What you are saying makes sense. Not enough for me to change my stance on my own functions, but since this is your function I'm fine with either choice. As long as you never write a function with name HJW your functions shouldn't collide with mine ;).

Reply:
Thanks again, much appreciated!